pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -178,55 +239,54 @@ func (p *ecrProvider) Enabled() bool {
 		Name: "k8s/logger",
 		Fn:   awsHandlerLogger,
 	})
-	p.getter = getter
+	return getter, nil
+}
 
-	return true
+// GetTokenGetterForRegion gets the token getter for the requested region. If it
+// doesn't exist, it creates a new ECR token getter
+func (f *ecrTokenGetterFactory) GetTokenGetterForRegion(region string) (tokenGetter, error) {
+	if getter, ok := f.cache[region]; ok {
+		return getter, nil
+	}
+	getter, err := newECRTokenGetter(region)
+	if err != nil {
+		klog.Errorf("while getting creating token getter for region %v %v", region, err)
```

Comment:
Should just be `while creating token getter` (extra "getting")

Reply:
IMO we should remove this log line and add this message as part of the returned error instead